overlord/devicestate/devicestate.go

```diff
@@ -1921,11 +1921,17 @@ func InstallFinish(st *state.State, label string, onVolumes map[string]*gadget.V
 	finishTask := st.NewTask("install-finish", fmt.Sprintf("Finish setup of run system for %q", label))
```

Comment:
might be worth already adding a TODO in the install-finish task handler about processing optional-install ?